fdbserver/ClusterController.actor.cpp

```diff
@@ -1691,6 +1700,196 @@ class ClusterControllerData {
 		return idUsed;
 	}
 
+	std::unordered_set<NetworkAddress> getServersWithLinkProblem() {
+		if (!SERVER_KNOBS->ENABLE_PEER_HEALTH_MONITOR) {
+			return {};
+		}
+
+		std::unordered_map<NetworkAddress, int> degradationCounts;
+		for (const auto& [server, degradedRemotePeers] : degradedPeers) {
+			for (const auto& [degradedServer, startTime] : degradedRemotePeers) {
+				if (now() - startTime < SERVER_KNOBS->PEER_DEGRADATION_INTERVAL) {
+					continue;
+				}
+				degradationCounts[degradedServer]++;
+			}
+		}
+
+		std::unordered_set<NetworkAddress> excludeServers;
+		for (const auto& [server, degradationCount] : degradationCounts) {
+			if (degradationCount <= SERVER_KNOBS->MAX_PEER_DEGRADATION_TO_EXCLUDE) {
+				excludeServers.insert(server);
```

Comment:
How about having a knob for dry-run?
CC prints which worker it intends to exclude but don't actually exclude it. 
It allows SRE to do sanity check before we enable automatic exclusion.